packages/devtools/src/transactions/format.ts

```diff
@@ -3,9 +3,12 @@ import { formatEid } from '@/omnigraph/format'
 
 export const formatOmniTransaction = (
     transaction: OmniTransaction
-): Record<string, string | number | bigint | undefined> => ({
+): Record<string, string | number | bigint | undefined | null> => ({
     Endpoint: formatEid(transaction.point.eid),
     OmniAddress: transaction.point.address,
+    OmniContract: transaction.point.contractName,
```

Comment:
This is always `undefined` because the transaction is built with [devtools-evm createTransaction](https://github.com/LayerZero-Labs/devtools/blob/5b1586fbab8f6f39860849e31b2cb8274c7c483a/packages/devtools-evm/src/omnigraph/sdk.ts#L59-L68), which uses [omniContractToPoint](https://github.com/LayerZero-Labs/devtools/blob/5b1586fbab8f6f39860849e31b2cb8274c7c483a/packages/devtools-evm/src/omnigraph/coordinates.ts#L5-L8) to build the point.

However, as you can see, the `omniContractToPoint` function does not specify the `contractName` -- only the `eid` and `address`, which I think is the root of the issue.

I have been looking into how to retrieve the contract name, but had no luck so far. It seems i need to do a reverse lookup on the ens registrar contract or use an rpc provider to call `lookupAddress`, both of which I do not have access to in the `devtools-evm/src/omnigraph/coordinates` file for `omniContractToPoint`.

Is there another way I can get the contract name?

Reply:
Oh yes [`omniContractToPoint`](https://github.com/LayerZero-Labs/devtools/blob/5b1586fbab8f6f39860849e31b2cb8274c7c483a/packages/devtools-evm/src/omnigraph/coordinates.ts#L5-L8)needs to be updated since the missing `contractName` property is a bug.

Reply:
I see, thank you for looking into this! @calebbanister please see Jan's explanation above.

While #869 is the long-term approach to fetch `contractName`, perhaps there is a quick "hacky" way to do it temporarily? It would really help when deploying/wiring stargate!

Reply:
@calebbanister mentioned we could:

> scrape the xxxxxxx-mainnet deployments folder and create a mapping of address -> contract name after splitting off the .json of the filenames

@janjakubnanista I can look into the above approach if you think there is no other "hacky way" to do it

Reply:
Please no ad-hoc mappings no. The fastest hacky way is to look into the contract factories - the factories have access to the original `OmniPoint` that contains the `contractName`. You can add custom constructors to SDKs whose names you want to log and pass this `OmniPoint` or just the `contractName`.